src/Model/CachingIterator.php

```diff
@@ -53,14 +53,14 @@ class CachingIterator implements Countable, Iterator
     /**
      * Initialize the iterator and stores the first item in the cache. This
      * effectively rewinds the Traversable and the wrapping IteratorIterator.
-     *  Additionally, this mimics behavior of the SPL iterators and allows users
-     * to omit an explicit call * to rewind() before using the other methods.
+     * Additionally, this mimics behavior of the SPL iterators and allows users
+     * to omit an explicit call to rewind() before using the other methods.
      *
      * @param Traversable $traversable
      */
     public function __construct(Traversable $traversable)
     {
-        $this->iterator = new IteratorIterator($traversable);
+        $this->iterator = $traversable instanceof Iterator ? $traversable : new IteratorIterator($traversable);
```

Comment:
Correct me if I'm wrong, but this could always have been done independent of this ticket (or PHPC 1.9.0), correct?

Seems like just a smarter change to avoid an extra IteratorIterator composition when we don't need it.

Reply:
That's correct, we could've always done this. I noticed this and made the change as most iterators passed now won't need to be wrapped as they already implement the `Iterator` interface.